src/voodoo/spellbook/spells/haste.ts

```diff
@@ -42,8 +42,9 @@ export const haste: SpellFunction = async (voodoo, accountId, upgradeConfigs) =>
 
   for (const playerId of playerIds) {
     const baseSpeed = await voodoo.getPlayerCheckStatBase({ accountId: playerId, stat: 'speed' });
+    const currentSpeed = await voodoo.getPlayerCheckStatCurrent({accountId: playerId, stat: 'speed' });
```

Comment:
Because this represents two consecutive calls to `player check-stat`, I'd introduce a new method that gets both of these in a single call.